### PR TITLE
(feat) Add asynchronous functionality to updateCalculatedValue() function

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
Add two main features.
1. Add ability to update controls when the calculated value is a promise and a helper funtion to fetch arbitraryinformation.

### CODE REVIEW

File: projects/ngx-formentry/src/abstract-controls-extension/afe-form-control.ts [hunk 1]
```diff
@@ -146,6 +146,7 @@ class AfeFormControl
   }
 
   setValue(value: any) {
+    console.log('setting value', value)
```
Comment (CynthiaKamau) [inline projects/ngx-formentry/src/abstract-controls-extension/afe-form-control.ts:149]: Kindly remove this

**Review by CynthiaKamau — APPROVED** [review]

File: projects/ngx-formentry/src/form-entry/question-models/question-base.ts [hunk 2]
```diff
@@ -42,6 +42,7 @@ export class QuestionBase implements BaseOptions {
   componentConfigs: Array<any>;
   options?: any;
   questionIndex?: Number;
+  message?: string;
```
Comment (ibacher) [inline projects/ngx-formentry/src/form-entry/question-models/question-base.ts:45]: I don't love having a message as a property of a question. How do we translate this?

**Review by ibacher — CHANGES_REQUESTED** [review]
So, I think this is ok from my perspective; however, we need to fix the places that rely on this behaviour being synchronous, specifically [here](https://github.com/openmrs/openmrs-ngx-formentry/blob/315970372d0a9b0ade2cf46dc68197cc11917e8c/projects/ngx-formentry/src/change-tracking/control-relation.ts#L34) and [here](https://github.com/openmrs/openmrs-ngx-formentry/blob/315970372d0a9b0ade2cf46dc68197cc11917e8c/projects/ngx-formentry/src/form-entry/form-factory/control-relations.factory.ts#L121) no longer function as expected because `updateCalculatedValue()` now returns a promise, rather than a value, which means that the subsequent lines of code need to be deferred until `updateCalculatedValue()` has resolved.

We should also update the signature [here](https://github.com/openmrs/openmrs-ngx-formentry/blob/315970372d0a9b0ade2cf46dc68197cc11917e8c/projects/ngx-formentry/src/form-entry/control-calculators/can-calculate.ts#L3) to reflect this change...

File: projects/ngx-formentry/src/form-entry/question-models/question-base.ts (hunk 2 above)
Comment (donaldkibet) [reply projects/ngx-formentry/src/form-entry/question-models/question-base.ts]: @ibacher removed the alert feature to be implemented in a follow up PR

**Review by CynthiaKamau — APPROVED** [review]

**Pull request closed without merging**